### DISCUSSION TITLE
Is there a way to prohibit usage of var keyword

### DISCUSSION THREAD

**Discussion by ChristianGmw** [TNG/ArchUnit#1051] [Q&A] [answered]
Hello, thanks for this amazing library!

In our project we have the rule that types should be explicit, so no usage of the Java 10 **"var"** keyword.

Is there a way to enforce this with ArchUnit?

**Comment by hankem** [comment, marked as answer]
As ArchUnit (currently) only analyzes byte code, it cannot distinguish between `var` and an explicit type declaration, as the compiler creates the same class file regardless of whether the type was expclitly declared or inferred.

**Reply by ChristianGmw** [reply]
Ah ok that makes sense! Then some sonar rule must do for now. ;-)

Thank you very much @hankem for the amazingly fast reply!
